src/brogue/Combat.c

```diff
@@ -1663,7 +1663,7 @@ void killCreature(creature *decedent, boolean administrativeDeath) {
         if (!administrativeDeath
             && decedent->creatureState == MONSTER_ALLY
             && !canSeeMonster(decedent)
-            && !(decedent->info.flags & MONST_INANIMATE)
+            && (!(decedent->info.flags & MONST_INANIMATE) || (decedent->info.monsterID == MK_PHOENIX_EGG))
```

Comment:
So the issue is that eggs are the only animate monster we want to resurrect?

Bear in mind this probably applies to Lich phylactories too, in the case one is dominated. I wonder if there should be another flag?

Reply:
Yeah there are other inanimate monsters like totems, stone/winged guardians, and spectral blades, but resurrection doesn't apply.

For some reason I thought there was something about liches where it wouldn't apply, but I just tested in wizard mode and you're right, it does apply. After dominating a lich, the ally status remains on both the phylactery and reborn lich, same as a phoenix. Maybe what threw me off was the fact that phylacteries take damage but phoenix eggs don't (unless negated).

I do think a flag would make sense. What do you think about a new monster ability flag MA_CAN_BE_RESURRECTED? It would need to be added to all applicable monsters in the catalog. Or should it be monster behavior flag? I'm not sure what difference is.

Reply:
Turns out we can use the existing flag MA_ENTER_SUMMONS, which is for phoenix eggs, phylacteries, and vampires. I also renamed MB_HAS_SOUL to MB_WEAPON_AUTO_ID and refactored status clearing/setting.